test/e2e/controller_test.go

```diff
@@ -218,4 +220,197 @@ var _ = Describe("Combination controller", func() {
+			GenerateName: "em-template",
+		},
+		Spec: v1alpha1.TemplateSpec{
+			Body:       rawTemplate,
+			Parameters: []string{"VALUE"},
+		},
+	}
+
+	embeddedCombo := v1alpha1.Combination{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "em-combination",
+		},
+		Spec: v1alpha1.CombinationSpec{
+			Arguments: []v1alpha1.Argument{
+				{
+					Key: "VALUE",
+					Values: []string{
+						"dev",
+						"prod",
+					},
+				},
+			},
+			Apply: true,
+		},
+	}
+
+	// This test ensures that when combo has permissions to create objects in a certain API group, it can do so successfully.
+	// Since by default combo only has permissions for its own APIs, this test uses a Template type in the template body
+	// so there are adequate permissions to create the resource on-cluster.
+	When("given a template with another template embedded in it that has apply set", func() {
+		var ctx context.Context
+
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			// before creation, modify the combo operator cluster role to allow for the creation of objects in the combo API group
+			// this simulates the cluster-admin modifying the permissions granted to combo before applying resources.
+			var cr rbacv1.ClusterRole
+			err := kubeclient.Get(ctx, types.NamespacedName{Name: "combo-operator"}, &cr)
+			Expect(err).To(BeNil(), "failed to get combo clusterrole")
+			cr.Rules[0].Verbs = append(cr.Rules[0].Verbs, "create")
+			err = kubeclient.Update(ctx, &cr)
+			Expect(err).To(BeNil(), "failed to update combo clusterrole")
+
+			err = kubeclient.Create(ctx, &embeddedTemplate)
+			Expect(err).To(BeNil(), "failed to create template CR")
+
+			embeddedCombo.Spec.Template = embeddedTemplate.Name
+			err = kubeclient.Create(ctx, &embeddedCombo)
+			Expect(err).To(BeNil(), "failed to create combo CR")
+		})
+
+		AfterEach(func() {
+			err := kubeclient.Delete(ctx, &embeddedTemplate)
+			Expect(err).To(BeNil(), "failed to cleanup template CR")
+
+			err = kubeclient.Delete(ctx, &embeddedCombo)
+			Expect(err).To(BeNil(), "failed to cleanup combo CR")
+		})
+
+		It("should successfully create the templated evaluation on-cluster", func() {
+			Eventually(func(g Gomega) error {
+				var retrievedCombination v1alpha1.Combination
+				err := kubeclient.Get(ctx, types.NamespacedName{Name: embeddedCombo.Name}, &retrievedCombination)
+				Expect(err).To(BeNil(), "failed to get combo CR")
+
+				var conditionReasons []string
+				for _, condition := range retrievedCombination.Status.Conditions {
+					conditionReasons = append(conditionReasons, condition.Reason)
+				}
+				g.Expect(conditionReasons).To(ContainElement(v1alpha1.ReasonProcessed))
+				// g.Expect(conditionReasons).To(ContainElement(v1alpha1.ReasonApplySucceeded))
+
+				// fetch created template on-cluster and ensure templating worked as expected
+				var embeddedTemplate v1alpha1.Template
+				err = kubeclient.Get(ctx, types.NamespacedName{Name: "embedded-template-dev"}, &embeddedTemplate)
```

Comment:
Handle error first, and then check whether the labels match our expectation? Alternatively, handle the error and updating the Eventually handler to check for the return value of the template labels?